codec-http2/src/main/java/io/netty/handler/codec/http2/PriorityStreamByteAssigner.java

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.http2;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+import static java.lang.Math.max;
+import static java.lang.Math.min;
+
+import java.util.Arrays;
+
+/**
+ * A {@link StreamByteAssigner} that implements the HTTP/2 priority tree algorithm for assigning
+ * bytes for all streams in the connection.
+ */
+public final class PriorityStreamByteAssigner implements StreamByteAssigner {
```

Comment:
I have found `assign` a few times in the RFC but not directly referencing "assigning bytes in flow control". However I have found stronger references using `allocate` in the rfc [1][2][3][4]. I agree that `allocate` can be a bit overloaded but it is a familiar term in this context (allocating resources). `assigner` seems like a new term potentially with its own overloaded meanings (assignment, other references to assign in the rfc, etc...). No strong preference but I wonder if introducing the new terminology is worth it here.

[1] https://tools.ietf.org/html/rfc7540#section-5.3
[2] https://tools.ietf.org/html/rfc7540#section-5.3.1
[3] https://tools.ietf.org/html/rfc7540#section-5.3.2
[4] https://tools.ietf.org/html/rfc7540#section-5.3.4